### PR TITLE
Modify headers to build projects with -pedantic gcc option

### PR DESCRIPTION
**Pull request by stac47**
Hello,
First of all thanks for this interesting library.
I needed to compile a project with the -pedantic gcc option but the fact you put semi-columns after each namespace block was a problem.
Regards,
Stac

### CODE REVIEW

**Comment by stac47** [comment]
Strange this failed. Can this be linked to the fact travis infrastructure changed ?

File: src/details/comet_manager.cpp [hunk 1]
```diff
@@ -25,11 +25,9 @@
 
 using namespace std;
 
-namespace httpserver
-{
+namespace httpserver {
```
Comment (etr) [inline src/details/comet_manager.cpp]: please avoid brackets on the same line

File: src/details/comet_manager.cpp [hunk 2]
```diff
@@ -25,11 +25,9 @@
...
 
-namespace details
-{
+namespace details {
```
Comment (etr) [inline src/details/comet_manager.cpp]: same here for the bracket

File: src/httpserver/binders.hpp [hunk 3]
```diff
@@ -26,349 +26,350 @@
 #define _BINDERS_HPP_
 
 namespace httpserver {
-namespace details
+namespace details {
```
Comment (etr) [inline src/httpserver/binders.hpp]: same here for the bracket

File: src/httpserver/binders.hpp [hunk 4]
```diff
@@ -26,349 +26,350 @@
...
+namespace binders {
```
Comment (etr) [inline src/httpserver/binders.hpp]: same here for the bracket

File: src/httpserver/details/cache_entry.hpp [hunk 5]
```diff
@@ -29,10 +29,8 @@
 #include <set>
 #include "httpserver/details/http_response_ptr.hpp"
 
-namespace httpserver
-{
-namespace details
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/details/cache_entry.hpp]: same here for the bracket

File: src/httpserver/details/cache_entry.hpp [hunk 6]
```diff
@@ -29,10 +29,8 @@
...
+namespace details {
```
Comment (etr) [inline src/httpserver/details/cache_entry.hpp]: same here for the bracket

File: src/httpserver/details/comet_manager.hpp [hunk 7]
```diff
@@ -33,13 +33,11 @@
 #include <string>
 #include "http_utils.hpp"
 
-namespace httpserver
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/details/comet_manager.hpp]: same here for the bracket

File: src/httpserver/details/comet_manager.hpp [hunk 8]
```diff
@@ -33,13 +33,11 @@
...
 
 class webserver;
 
-namespace details
-{
+namespace details {
```
Comment (etr) [inline src/httpserver/details/comet_manager.hpp]: same here for the bracket

File: src/httpserver/details/http_endpoint.hpp [hunk 9]
```diff
@@ -30,14 +30,12 @@
 #include <regex.h>
 #include <string>
 
-namespace httpserver
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/details/http_endpoint.hpp]: same here for the bracket

File: src/httpserver/details/http_endpoint.hpp [hunk 10]
```diff
@@ -30,14 +30,12 @@
...
 
 class webserver;
 class http_resource;
 
-namespace details
-{
+namespace details {
```
Comment (etr) [inline src/httpserver/details/http_endpoint.hpp]: same here for the bracket

File: src/httpserver/details/http_response_ptr.hpp [hunk 11]
```diff
@@ -45,13 +45,11 @@
 
 #endif
 
-namespace httpserver
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/details/http_response_ptr.hpp]: same here for the bracket

File: src/httpserver/details/http_response_ptr.hpp [hunk 12]
```diff
@@ -45,13 +45,11 @@
...
 
 class webserver;
 
-namespace details
-{
+namespace details {
```
Comment (etr) [inline src/httpserver/details/http_response_ptr.hpp]: same here for the bracket

File: src/httpserver/details/modded_request.hpp [hunk 13]
```diff
@@ -28,11 +28,8 @@
 #include "binders.hpp"
 #include "details/http_response_ptr.hpp"
 
-namespace httpserver
-{
-
-namespace details
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/details/modded_request.hpp]: same here for the bracket

File: src/httpserver/details/modded_request.hpp [hunk 14]
```diff
@@ -28,11 +28,8 @@
...
+namespace details {
```
Comment (etr) [inline src/httpserver/details/modded_request.hpp]: same here for the bracket

File: src/httpserver/http_request.hpp [hunk 15]
```diff
@@ -33,16 +33,16 @@
 
 struct MHD_Connection;
 
-namespace httpserver
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/http_request.hpp]: same here for the bracket

File: src/httpserver/http_request.hpp [hunk 16]
```diff
@@ -33,16 +33,16 @@
...
 
 class webserver;
 
-namespace http
-{
-    class header_comparator;
-    class arg_comparator;
-};
+namespace http {
```
Comment (etr) [inline src/httpserver/http_request.hpp]: same here for the bracket

File: src/httpserver/http_resource.hpp [hunk 17]
```diff
@@ -31,19 +31,17 @@
 #include <iostream>
 #endif
 
-namespace httpserver
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/http_resource.hpp]: same here for the bracket

File: src/httpserver/http_resource.hpp [hunk 18]
```diff
@@ -31,19 +31,17 @@
...
 
 class webserver;
 class http_request;
 class http_response;
 
-namespace details
-{
+namespace details {
```
Comment (etr) [inline src/httpserver/http_resource.hpp]: same here for the bracket

File: src/httpserver/http_response.hpp [hunk 19]
```diff
@@ -35,24 +35,21 @@
 
 struct MHD_Connection;
 
-namespace httpserver
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/http_response.hpp]: same here for the bracket

File: src/httpserver/http_response.hpp [hunk 20]
```diff
@@ -35,24 +35,21 @@
...
 
 class webserver;
 class http_response_builder;
 
-namespace http
-{
+namespace http {
```
Comment (etr) [inline src/httpserver/http_response.hpp]: same here for the bracket

File: src/httpserver/http_response.hpp [hunk 21]
```diff
@@ -35,24 +35,21 @@
...
     class header_comparator;
     class arg_comparator;
-};
+} // namespace http
 
-namespace details
-{
+namespace details {
```
Comment (etr) [inline src/httpserver/http_response.hpp]: same here for the bracket

File: src/httpserver/http_response_builder.hpp [hunk 22]
```diff
@@ -30,20 +30,17 @@
 
 struct MHD_Connection;
 
-namespace httpserver
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/http_response_builder.hpp]: same here for the bracket

File: src/httpserver/http_response_builder.hpp [hunk 23]
```diff
@@ -30,20 +30,17 @@
...
 
 class webserver;
 
-namespace http
-{
+namespace http {
```
Comment (etr) [inline src/httpserver/http_response_builder.hpp]: same here for the bracket

File: src/httpserver/http_response_builder.hpp [hunk 24]
```diff
@@ -30,20 +30,17 @@
...
     class header_comparator;
-};
+} // namespace http
 
-namespace details
-{
+namespace details {
```
Comment (etr) [inline src/httpserver/http_response_builder.hpp]: same here for the bracket

File: src/httpserver/string_utilities.hpp [hunk 25]
```diff
@@ -28,10 +28,8 @@
 #include <string>
 #include <vector>
 
-namespace httpserver
-{
-namespace string_utilities
-{
+namespace httpserver {
```
Comment (etr) [inline src/httpserver/string_utilities.hpp]: same here for the bracket

File: src/httpserver/string_utilities.hpp [hunk 26]
```diff
@@ -28,10 +28,8 @@
...
+namespace string_utilities {
```
Comment (etr) [inline src/httpserver/string_utilities.hpp]: same here for the bracket

**Comment by etr** [comment]
That wasn't your fault - there were problems due to the use of atomics using functions non implemented in gcc4.6 (the one used by travis).

Everything seems alright with the change apart for all the brackets in opening of namespaces - the brackets are inlined in your change whereas the standard for this library is to have them on separate line.

**Comment by stac47** [comment]
Ok no problem, I did this slight change regarding brackets.
Regarding the atomics, why would we migrate the lib to C++11 to use the C++ standard lib atomics API ?

**Comment by etr** [comment]
Many people are currently using the library on systems that do not support versions of gcc that are new enough to have C++11. I need to make an assessment on this before removing backward compatibility support on such a low level library.

Two brackets (in comet_manager) are still reported by the diff as being misplaced. Fixed this, everything seems ok.

**Comment by aggsol** [comment]
> 
> 
> Many people are currently using the library on systems that do not support versions of gcc that are new enough to have C++11. I need to make an assessment on this before removing backward compatibility support on such a low level library.
> 
> Two brackets (in comet_manager) are still reported by the diff as being misplaced. Fixed this, everything seems ok.

Did you come to a conclusion now? Maybe C++14 or even C++17 is now an option?

**Comment by etr** [comment]
pedantic is now checked when building the library with debug on through validation

**Pull request closed without merging**